src/State/Scope.php

```diff
@@ -96,6 +102,38 @@ public function getTags(): array
         return $this->tags->toArray();
     }
 
+    /**
+     * Sets context data with the given name.
+     *
+     * @param string $name  The name that uniquely identifies the context
+     * @param array  $value The value
+     *
+     * @return self
+     */
+    public function setContext(string $name, array $value): self
+    {
+        if (!isset($this->contexts[$name])) {
+            $this->contexts[$name] = new \stdClass();
```

Comment:
I don't really like to have to use `stdClass`, what's the point of doing it instead of using `$value` directly?

Reply:
Sentry only accepts objects for `contexts`, any other value will make an error..

Reply:
This is reasonable only if the user is using numeric-only keys, so `json_encode` will mistake it for an array.

Reply:
@wedamija you're right, but there is only one case that cames  to my mind that could cause PHP to JSON-encode in the wrong format the array: when it represents a pure list (so 0-based index). In all other cases it should work, and I expect people to use the context more as a map than a list. Of course this cannot be strictly enforced, we could add a check like in the `TagsContext` that the key is a string until PHP will have true generics that will allow typehinting the arrays

Reply:
@ste93cry What should I do ?

Reply:
> we could add a check like in the TagsContext that the key is a string

I'd be in favor of doing so as well.

Reply:
@ste93cry do I need to make a change here ?

as I said "Sentry only accepts objects for contexts, any other value will make an error."

Reply:
>do I need to make a change here ?

Yes please, let's start with `array` and let's see what happens and how the situation evolves in the future. Also please update the code to use a plain array for the `contexts` variable and add the `setContexts` method to the `Event` class.

>"Sentry only accepts objects for contexts, any other value will make an error."

You're right, however it's unlikely that someone will put a list as value of a context and it's unlikely that someone will use `string` keys containing numbers knowing that they will be casted

Reply:
Okay, made the required changes (rebase, amend fix)
See 2419b39aa95b328af9e9a8bf12ba5387bc330618